File: contracts/wasm/myworld/go/myworld/myworld.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package myworld

import "github.com/iotaledger/wasp/packages/vm/wasmlib/go/wasmlib"

func funcDepositTreasure(ctx wasmlib.ScFuncContext, f *DepositTreasureContext) {
}

func funcInit(ctx wasmlib.ScFuncContext, f *InitContext) {
	if f.Params.Owner().Exists() {
		f.State.Owner().SetValue(f.Params.Owner().Value())
		return
	}
	f.State.Owner().SetValue(ctx.ContractCreator())
}

func funcSetOwner(ctx wasmlib.ScFuncContext, f *SetOwnerContext) {
	f.State.Owner().SetValue(f.Params.Owner().Value())
}

func viewGetAllTreasures(ctx wasmlib.ScViewContext, f *GetAllTreasuresContext) {
}

func viewGetOwner(ctx wasmlib.ScViewContext, f *GetOwnerContext) {
	f.Results.Owner().SetValue(f.State.Owner().Value())
}

File: contracts/wasm/myworld/src/myworld.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use wasmlib::*;

use crate::*;
use crate::structs::*;

pub fn func_deposit_treasure(ctx: &ScFuncContext, f: &DepositTreasureContext) {
}

pub fn func_init(ctx: &ScFuncContext, f: &InitContext) {
    if f.params.owner().exists() {
        f.state.owner().set_value(&f.params.owner().value());
        return;
    }
    f.state.owner().set_value(&ctx.contract_creator());
}

pub fn func_set_owner(ctx: &ScFuncContext, f: &SetOwnerContext) {
    f.state.owner().set_value(&f.params.owner().value());
}

pub fn view_get_all_treasures(ctx: &ScViewContext, f: &GetAllTreasuresContext) {
}

pub fn view_get_owner(ctx: &ScViewContext, f: &GetOwnerContext) {
    f.results.owner().set_value(&f.state.owner().value());
}
